apollo-router-core/src/json_ext.rs

```diff
@@ -35,94 +26,30 @@ pub trait ValueExt {
-                            .as_array()
-                            .ok_or(JsonExtError::PathNotFound)?
-                            .get(*i)
-                            .ok_or(JsonExtError::PathNotFound),
-                        (PathElement::Flatten, _) => unreachable!(),
-                    })
-                    .collect::<Result<Vec<_>, _>>()?,
-            }
-        }
+    /// Create a `Value` by inserting a value at a subpath.
+    ///
+    /// This will create objects, arrays and null nodes as needed if they
+    /// are not present: the resulting Value is meant to be merged with an
+    /// existing one that contains those nodes.
+    #[track_caller]
+    fn from_path(path: &Path, value: Value) -> Value;
 
-        Ok(current)
-    }
-
-    fn get_at_path_mut<'a, 'b>(
-        &'a mut self,
-        path: &'b Path,
-    ) -> Result<Vec<&'a mut Self>, JsonExtError> {
-        let mut current = vec![self];
-
-        for path_element in path.iter() {
-            current = match path_element {
-                PathElement::Flatten => current
-                    .into_iter()
-                    .map(|x| x.as_array_mut().ok_or(JsonExtError::InvalidFlatten))
-                    .collect::<Result<Vec<_>, _>>()?
-                    .into_iter()
-                    .flatten()
-                    .collect::<Vec<_>>(),
-                PathElement::Index(index) if !(current.len() == 1 && current[0].is_array()) => {
-                    vec![current
-                        .into_iter()
-                        .nth(*index)
-                        .ok_or(JsonExtError::PathNotFound)?]
-                }
-                path_element => current
-                    .into_iter()
-                    .map(|value| match (path_element, value) {
-                        (PathElement::Key(key), value) if value.is_object() => value
-                            .as_object_mut()
-                            .unwrap()
-                            .get_mut(key)
-                            .ok_or(JsonExtError::PathNotFound),
-                        (PathElement::Key(_), _) => Err(JsonExtError::PathNotFound),
-                        (PathElement::Index(i), value) => value
-                            .as_array_mut()
-                            .ok_or(JsonExtError::PathNotFound)?
-                            .get_mut(*i)
-                            .ok_or(JsonExtError::PathNotFound),
-                        (PathElement::Flatten, _) => unreachable!(),
-                    })
-                    .collect::<Result<Vec<_>, _>>()?,
-            }
-        }
+    /// Insert a `value` at a `Path`
+    #[track_caller]
+    fn insert(&mut self, path: &Path, value: Value) -> Result<(), FetchError>;
 
-        Ok(current)
-    }
+    /// Select all values matching a `Path`.
+    ///
+    /// this will return the values and their specific path in the `Value`:
+    /// a `Path` with a flatten node can match multiple values.
```

Comment:
:yarn: This comment is outdated as this function doesn't return much.

Reply:
fixed in 1e5489c